### PR TITLE
refactor: move isAction to compass-utils

### PR DESCRIPTION
**Pull request by mabaasit**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-utils/src/index.ts [hunk 1]
```diff
@@ -1,5 +1,3 @@
-export type { AmpersandMethodOptions } from './promisify-ampersand-method';
-export { promisifyAmpersandMethod } from './promisify-ampersand-method';
```
Comment (mabaasit) [inline packages/compass-utils/src/index.ts:2]: drive-by: we are not using these anywhere, so cleaned it up

File: packages/compass-utils/src/is-action.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Types duplicated from redux.
+interface Action<T = any> {
+  type: T;
+}
+
+interface AnyAction extends Action {
+  [extraProps: string]: any;
+}
+
+export function isAction<A extends AnyAction>(
```
Comment (mabaasit) [inline packages/compass-utils/src/is-action.ts:10]: or should i rename it to `isReduxAction`?

File: packages/atlas-service/src/store/atlas-signin-reducer.ts [hunk 3]
```diff
@@ -3,14 +3,7 @@ import type { ThunkAction } from 'redux-thunk';
...
 import type { AtlasAuthService } from '../provider';
-import { throwIfAborted } from '@mongodb-js/compass-utils';
-
-export function isAction<A extends AnyAction>(
-  action: AnyAction,
-  type: A['type']
-): action is A {
-  return action.type === type;
-}
+import { throwIfAborted, isAction } from '@mongodb-js/compass-utils';
```
[the remark this answers is not in the document]
Comment (mcasimir) [reply packages/atlas-service/src/store/atlas-signin-reducer.ts:6]: > have a completely separate package for these redux helpers

@gribnoysup any clearer idea on how to evolve things in this context? Is parking redux helpers in `compass-utils` ok as a general practice, or would mud even further the purpose of this package?

[the remark this answers is not in the document]
Comment (mabaasit) [reply packages/atlas-service/src/store/atlas-signin-reducer.ts:6]: i am not sure what will come out of `compass-redux` project, but i guess we may have a package for all the redux helpers we have to do everywhere for - `isAction`,  store specific`ThunkAction`and `ThunkDispatch`. 

currently we are using `throwIfAborted` in atlas-service which means that this dependency issue is already there and we should split electron helpers to a named export from this irrespective of this PR.

[the remark this answers is not in the document]
Comment (mcasimir) [reply packages/atlas-service/src/store/atlas-signin-reducer.ts:6]: Makes total sense @mabaasit thanks! and is a great suggestion going forward.

File: packages/compass-utils/src/is-action.ts (hunk 2 above)
Comment (addaleax) [reply packages/compass-utils/src/is-action.ts:10]: Just my 2 cents, but I think we're going to be fine with sticking to `isAction` unless we plan on introducing yet another state management library with its own `Action` concept 🙂
